packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -161,26 +154,31 @@ function routerReducer(
 			screens = removeScreen( state, action.screen );
 			break;
 		case 'goto':
-			locationHistory = goTo( state, action.path, action.options );
+			const goToNewState = goTo( state, action.path, action.options );
+			currentLocation = goToNewState.currentLocation;
+			focusSelectors = goToNewState.focusSelectors;
```

Comment:
You can do a destructuring assignment in only you wrap it in parens so that it's not confused with a block:
```js
( { currentLocation, focusSelection } = goTo() );
```